SpatialGDK/Source/SpatialGDK/Private/EngineClasses/Components/SpatialPingComponent.cpp

```diff
@@ -181,6 +182,48 @@ void USpatialPingComponent::OnRep_ReplicatedPingID()
 	}
 }
 
+FSpatialPingAverageData USpatialPingComponent::GetAverageData() const
+{
+	FSpatialPingAverageData Data = {};
+
+	if (LastPingMeasurements.Num() > 0)
+	{
+		double Total = 0.0;
+		for (float Ping : LastPingMeasurements)
+		{
+			Total += Ping;
+		}
+		Data.LastMeasurementsWindowAvg = Total / LastPingMeasurements.Num();
+		Data.LastMeasurementsWindowMin = FMath::Min(LastPingMeasurements);
+		Data.LastMeasurementsWindowMax = FMath::Max(LastPingMeasurements);
+	}
+	Data.WindowSize = LastPingMeasurements.Num();
+
+	Data.TotalAvg = TotalPing / TotalNum;
+	Data.TotalMin = TotalMin;
+	Data.TotalMax = TotalMax;
+	Data.TotalNum = TotalNum;
+
+	return Data;
+}
+
+void USpatialPingComponent::RecordPing(float Ping)
+{
+	LastPingMeasurements.Add(Ping);
+	if (LastPingMeasurements.Num() > PingMeasurementsWindowSize)
+	{
+		LastPingMeasurements.RemoveAt(0);
+	}
+
+	TotalPing += Ping;
```

Comment:
Do we need to - LastPingMeasurements[0] before it was removed? Is this the total ping of the sliding window?

Reply:
No, this is all time total.